core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -478,4 +492,22 @@ private SortOption analyzeSortOption(List<Argument> fieldArgs) {
     return asc ? SortOption.DEFAULT_ASC : SortOption.DEFAULT_DESC;
   }
 
+  private String getCatalogNameFromQualifiedTableFunctionName(String qualifiedFunctionName) {
```

Comment:
we should have logic to clear define how to extract catalog name from identifier. Currently, Relation has getCatalogName() already. 
@MaxKsyunz bring up this topic also. https://github.com/opensearch-project/sql/discussions/871

Reply:
Yes...will look into the discussion and refactor.

Reply:
Will change this logic....with this PR.
https://github.com/opensearch-project/sql/pull/876/files